### ISSUE TITLE
Custom Diffusion Delta values for R017pre6 for VE12U-AP02

### ISSUE THREAD

**Issue by whitecraine** [CMRR-C2P/MB#362]
Is there a way to reduce the min delta and DELTA values (e.g., via MBAdvancedSettings.ini)? The smallest values (even with b-value set to near 0) is 7.66 ms and 23.9 ms respectively. It would be great to have access to shorter diffusion encoding times also for shorter TEs (up to around 20 ms shorter). 

Thank you!

Joseph

**Comment by eauerbach** [comment]
If you are running at minimum TE they should be already as short as possible based on the system configuration, although in the case of special systems the preconfigured parameters may not be appropriate. The default maximum gradient amplitude is 95% of the defined absolute maximum, and the default risetime is 16 us/(mT/m). There are some specific exceptions to this, e.g. Prisma uses 20 us/(mT/m) and the old Allegra head gradients used 10 us/(mT/m). Probably you will need to optimize this empirically (at your own risk!).

The delta is optimized based on the gradient maximum amplitude and ramp. In the latest version you can see the maximum configured and used gradient amplitudes in the tooltip over the diffusion mode selection box. DELTA is additionally affected by the refocusing RF pulse length.

There is a way to override the diffusion gradient maximum amplitude and rise time by editing the file %CustomerSeq%\SBBDiffusion.ini with parameters like this (note that this may also affect the product diffusion sequence):

```
[SBBDiffusion_Base::SBBDiffusion_Base]
MaxGradAmpl = 50.0
RiseTime = 16.0
```

**Comment by mharms** [comment]
The output of `imprint` indicates that `GradMinRiseTime` for a Prisma in "Performance" mode is 5.3 us/(mT/m). Is that not relevant to the _diffusion gradients_, which are perhaps fixed at 20 us/(mT/m) [i.e., slew rate of 50 (T/m)/s] regardless of the selected Gradient Mode?

**Comment by eauerbach** [comment]
Correct, the Whisper/Normal/Fast/Performance gradient mode parameters are ignored for the diffusion encoding gradients. The diffusion encoding gradients are a special case and always use 95% of GradMaxAmplAbsolute for maximum amplitude and 16 us/(mT/m) for rise time—with a few system-specific exceptions. One of those exceptions is for Prisma the rise time is 20 us/(mT/m).

**Comment by tkim2021** [comment]
where can I find MBAdvancedSettings.ini? How can I find min delta and DELTA values for cmrr diffusion sequence in Prisma (VE11C or VE11E)?

**Comment by eauerbach** [comment]
UI options for controlling delta/DELTA have been added in R017pre13 (see the Wiki for the ini file option that needs to be set to unlock them). Note that when these options are enabled it is likely the rest of the UI will behave badly so use with care.

**Comment by whitecraine** [comment]
Thank you for adding this. Unfortuantely, when I tried adding it to the .ini file the checkbox is grayed out no matter what other changes I made to the other paramters. Any suggestions?

![Image](https://github.com/user-attachments/assets/b7f16bd4-53eb-4e84-b0bc-b6ac34565edd)

**Comment by eauerbach** [comment]
I would suggest starting from a new default protocol. I can only verify it works fine in the simulator for a new default protocol (you do have to change the diffusion mode to Monopolar first for it to appear). The default values for delta/DELTA are 40ms and 100ms, respectively, so you need to start with rather long TE/TR. The UI solve handlers do not work for these parameters so it will not offer to fix anything if the protocol is invalid for any reason; things will just get grayed out.

_Maybe_ you can adapt an old protocol to work if you find the right combination of parameters, but it is also possible an old protocol may not have default values for delta/DELTA populated at all (since I never really expected for this option to be generally available I did not include the logic to update the defaults when converting/upgrading old protocols). In that case maybe it is trying to use default values of 0 which will never be valid.

**Comment by whitecraine** [comment]
> If you are running at minimum TE they should be already as short as possible based on the system configuration, although in the case of special systems the preconfigured parameters may not be appropriate. The default maximum gradient amplitude is 95% of the defined absolute maximum, and the default risetime is 16 us/(mT/m). There are some specific exceptions to this, e.g. Prisma uses 20 us/(mT/m) and the old Allegra head gradients used 10 us/(mT/m). Probably you will need to optimize this empirically (at your own risk!).
> 
> The delta is optimized based on the gradient maximum amplitude and ramp. In the latest version you can see the maximum configured and used gradient amplitudes in the tooltip over the diffusion mode selection box. DELTA is additionally affected by the refocusing RF pulse length.
> 
> There is a way to override the diffusion gradient maximum amplitude and rise time by editing the file %CustomerSeq%\SBBDiffusion.ini with parameters like this (note that this may also affect the product diffusion sequence):
> 
> ```
> [SBBDiffusion_Base::SBBDiffusion_Base]
> MaxGradAmpl = 50.0
> RiseTime = 16.0
> ```

Hi Eddy, can you confirm if these instructions apply to XA61_sp1 as well? We tried adding the %CustomerSeq%\SBBDiffusion.ini but the reported risetime in the diffusion tool tip didn't change.

**Comment by eauerbach** [comment]
That ini file no longer exists in NX. The maximum amplitude parameter can still be set using the debug settings tool: EP2D_DIFF/max_grad_ampl

Unfortunately the manual diffusion gradient risetime parameter no longer exists at all. I suppose I could add it back in, but since XA60A there is a new dynamic optimization ("StimulationChecker") that would just override it later. I do have an option to disable that optimization, which is sometimes useful to match old Terra protocols on Terra.X W60, but it would probably make Cima.X unusable. I could maybe add a lower bound (MinRiseTime) if that is useful.

**Comment by whitecraine** [comment]
Thank you for looking into this Eddie. For the XA60A platform, it seems the diffusion ramp time is fixed to 7.6ms for the CMRR sequence, independent of the b-value and gradient strength needed. The Siemens Diffusion Advanced WIP can get closer to 1 ms in just one an example use case. This results in unusably long TEs (TE=39 ms for Diffusion Advanced WIP vs TE = 76 for CMRR). Is it possible there will be some hints from Siemens on how to customize the ramp time and rise time based on their WIP implementation? 

> That ini file no longer exists in NX. The maximum amplitude parameter can still be set using the debug settings tool: EP2D_DIFF/max_grad_ampl
> 
> Unfortunately the manual diffusion gradient risetime parameter no longer exists at all. I suppose I could add it back in, but since XA60A there is a new dynamic optimization ("StimulationChecker") that would just override it later. I do have an option to disable that optimization, which is sometimes useful to match old Terra protocols on Terra.X W60, but it would probably make Cima.X unusable. I could maybe add a lower bound (MinRiseTime) if that is useful.
